crates/typed-store/src/metrics.rs

```diff
@@ -1,12 +1,72 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
-use std::sync::Arc;
-
 use once_cell::sync::OnceCell;
 use prometheus::{
     exponential_buckets, register_histogram_vec_with_registry,
     register_int_counter_vec_with_registry, HistogramVec, IntCounterVec, Registry,
 };
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+
+#[derive(Debug, Clone)]
+// A struct for sampling based on number of operations or duration.
+// Sampling happens if the duration expires and after number of operations
+pub struct SamplingInterval {
+    // Sample once every time duration
+    pub once_every_duration: Duration,
+    // Sample once every number of operations
+    pub after_num_ops: u64,
+    // Counter for keeping track of previous sample
+    pub counter: Arc<AtomicU64>,
```

Comment:
Here @sadhansood , you use an Arc of Atomic here for counter.  However if SamplingInterval is purely used within `ThreadLocal` then Arc of Atomic is not necessary.

I think you used to have an atomic for the time interval too but you must have cleaned that up.

Reply:
@velvia - SamplingInterval is not used within any threadlocal. The perf context is thread local, however we need to sample enabling perf context across multiple threads (and hence thread safe SamplingInterval is needed)